back/src/domains/convention/entities/AssessmentEntity.ts

```diff
@@ -1,26 +1,55 @@
-import { AssessmentDto, ConventionDto, ConventionStatus, errors } from "shared";
+import {
+  AssessmentDto,
+  ConventionDto,
+  ConventionStatus,
+  calculateTotalImmersionHoursBetweenDateComplex,
+  errors,
+} from "shared";
 import { EntityFromDto } from "../../../utils/EntityFromDto";
 
-export type AssessmentEntity = EntityFromDto<AssessmentDto, "Assessment">;
+export type AssessmentEntity = EntityFromDto<AssessmentDto, "Assessment"> & {
+  numberOfHoursActuallyMade: number | null;
+};
 
 export const acceptedConventionStatusesForAssessment: ConventionStatus[] = [
   "ACCEPTED_BY_VALIDATOR",
 ];
 
+const calculateNumberOfHoursActuallyMade = (
+  assessment: AssessmentDto,
+  convention: ConventionDto,
+): number => {
+  if (assessment.status === "DID_NOT_SHOW") return 0;
+  if (assessment.status === "COMPLETED") return convention.schedule.totalHours;
+
+  return (
+    calculateTotalImmersionHoursBetweenDateComplex({
+      complexSchedule: convention.schedule.complexSchedule,
+      dateStart: convention.dateStart,
+      dateEnd: assessment.lastDayOfPresence ?? convention.dateEnd,
+    }) - assessment.numberOfMissedHours
+  );
```

Comment:
pour s'assurer que ça ne donne pas un nombre à virgule à rallonge 
```suggestion
  return (
    calculateTotalImmersionHoursBetweenDateComplex({
      complexSchedule: convention.schedule.complexSchedule,
      dateStart: convention.dateStart,
      dateEnd: assessment.lastDayOfPresence ?? convention.dateEnd,
    }) - assessment.numberOfMissedHours
  ).toFixed(2);
```